datasafe-storage/datasafe-storage-impl-s3/src/test/java/de/adorsys/datasafe/business/impl/storage/MultipartUploadS3SystemStorageServiceTest.java

```diff
@@ -0,0 +1,108 @@
+package de.adorsys.datasafe.business.impl.storage;
+
+import de.adorsys.datasafe.business.api.types.resource.AbsoluteLocation;
+import de.adorsys.datasafe.business.api.types.resource.BasePrivateResource;
+import de.adorsys.datasafe.business.api.types.resource.PrivateResource;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.platform.commons.util.StringUtils;
+import org.testcontainers.shaded.org.bouncycastle.util.encoders.Hex;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@Slf4j
+public class MultipartUploadS3SystemStorageServiceTest extends S3SystemStorageServiceTest {
+
+    @TempDir
+    protected Path tempDir;
+
+    @Test
+    @SneakyThrows
+    void write() {
+        String testFileName = tempDir.toString() + "/test.txt";
+
+        int loadS3TestFileSizeMb = 100;//1024;//1;
+
+        String loadS3TestfileSizeMb = System.getenv("LOAD_S3_TESTFILE_SIZE_MB");
+        if(StringUtils.isNotBlank(loadS3TestfileSizeMb)) {
+            loadS3TestFileSizeMb = Integer.parseInt(loadS3TestfileSizeMb);
+        }
+
+        generateTestFile(testFileName, loadS3TestFileSizeMb);
+
+        String testFileChecksum = calculateChecksumOfTestFile(testFileName);
+
+        AbsoluteLocation<PrivateResource> privateLocation = new AbsoluteLocation<>(
+                BasePrivateResource.forPrivate(URI.create("s3://" + bucketName + "/bigfile.txt")));
+
+        try (OutputStream os = storageService.write(privateLocation)) {
+            int _1mb = 1024 * 1024;
+            for (int i = 1; i <= loadS3TestFileSizeMb; i++) {
+                os.write(generateArrayWithSize(_1mb));
```

Comment:
See de.adorsys.datasafe.business.impl.e2e.performance.services.ContentGenerator and its usages
```
try (OutputStream os = openWriteStream(user, oper)) {
            ByteStreams.copy(user.getGenerator().generate(oper.getContentId().getId()), os);
        }
``` 
it won't use any significant memory amount at all and you can move it to testshared